spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/pubsub/GcpPubSubEmulatorAutoConfiguration.java

```diff
@@ -39,6 +40,7 @@
  * @author Mike Eltsufin
  */
 @Configuration(proxyBeanMethods = false)
+@ConditionalOnClass(ManagedChannel.class)
```

Comment:
There is a potential problem with using `ManagedChannel` because it's not unique to Pub/Sub and could be brought in by another GCP dependency. I think this class should actually be conditional on the same as `GcpPubSubAutoConfiguration`.

Would this work?
```suggestion
@ConditionalOnProperty(value = "spring.cloud.gcp.pubsub.enabled", matchIfMissing = true)
@ConditionalOnClass(PubSubTemplate.class)
```

Reply:
Yes, that makes sense. Though theoretically I guess we could have a `PubsubTemplate` on the classpath and not a `ManagedChannel`, which would cause this to break again. 
Maybe we should add `PubSubTemplate` to the list instead of removing  `@ConditionalOnClass(ManagedChannel.class)`? 
```
@ConditionalOnClass({ManagedChannel.class, PubSubTemplate.class})
```

Reply:
Sure, that works too. Although, gRPC is a core dependency of Pub/Sub, and that's where the `ManagedChannel` is coming from.

Reply:
Yeah, something would be really off if `ManagedChannel` is not on the classpath when `PubSubTemplate` is, but I guess it could happen.

Added it in ba7464ecae57cd897519b0878d7ef74c8a371e86

Reply:
I think we also might want `@ConditionalOnProperty(value = "spring.cloud.gcp.pubsub.enabled", matchIfMissing = true)` because there is no reason for this autoconfig to run when Pub/Sub autoconfig is disabled. Right?

Reply:
Totally agree - added in b195f4d4 :+1: